hack/install-vm.sh

```diff
@@ -141,17 +150,238 @@ EOF
+        execute_command "$cmd" 1m
+        if [ "$?" -eq 0 ]; then
+            echo "docker-compose is downloaded successfully"
+            chmod +x ${DOCKER_COMPOSE_DIR}
+            ln -s ${DOCKER_COMPOSE_DIR} /usr/bin/docker-compose
+        else
+            echoerr "Error docker-compose couldn't be downloaded, exiting"
+            exit 2
+        fi
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with ovs and antrea-agent container specific parameters.
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    cap_add:
+        - NET_ADMIN
+        - SYS_NICE
+        - SYS_ADMIN
+        - IPC_LOCK
+
+  antrea-agent:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-agent
+    volumes:
+        - /etc/antrea/:/etc/antrea
+        - /var/run/antrea:/var/run/antrea
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /var/lib/cni:/var/lib/cni
+        - /var/log/antrea:/var/log/antrea
+        - /proc:/host/proc
+        - /var/run/netns:/host/var/run/netns
+        - /run/xtables.lock:/run/xtables.lock
+    environment:
+        - NODE_NAME=$NODE_NAME
+    command: $ANTREA_AGENT_BIN --config $AGENT_CONF_PATH/antrea-agent.conf --logtostderr=false --log_file=$AGENT_LOG_DIR/antrea-agent.log
+    network_mode: host
+    privileged: true
+    cap_add:
+        - SYS_MODULE
+EOF
+}
+
+create_antrea_agent_service_file() {
+    cat >$AGENT_CONF_PATH/antrea-agent.sh << 'EOF'
+#!/bin/bash
+
+AGENT_CONF_PATH="/etc/antrea"
+AGENT_LOG_DIR="/var/log/antrea"
+COUNT=0
+LIMIT=1000
+
+declare -a OVS_KERNEL_MODULES=("vport_geneve" "openvswitch")
+declare -a CONTAINERS=("ovs" "antrea-agent")
+
+load_ovs_modules() {
+    echo "Loading OVS kernel"
+    for module in "${OVS_KERNEL_MODULES[@]}"; do
+        lsmod | grep -q $module
+        if [[ $? -ne 0 ]]; then
+            echo "Load $module kernel module"
+            modprobe $module
+            if [ "$?" -ne 0 ]; then
+                echoerr "Error load $module kernel module"
+            fi
+        fi
+    done
+}
+
+exec > >(tee $AGENT_LOG_DIR/vm-watchdog.log) 2>&1
+
+docker info
+if [[ $? -ne 0 ]]; then
+    echo "Starting docker service"
+    systemctl start docker
+    sleep 2
+fi
+
+load_ovs_modules
+
+while [ $COUNT -lt $LIMIT ] ; do
+    for service in "${CONTAINERS[@]}"; do
+        if [ ! "$(docker ps -q -f name=$service)" ]; then
+            if [ "$(docker ps -aq -f status=exited -f name=$service)" ]; then
+                echo "$service is in exited state"
+                ((COUNT=COUNT+1))
+            fi
+            echo "Starting containers"
```

Comment:
Dump the variable CONTAINERS in the log message.

Reply:
Done.